academic-services/knowledge-exploration-service/reference-index-data.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Entity data files
+description: Describes the format for defining entity data to be indexed
+ms.topic: reference
+ms.date: 7/1/2020
+---
+
+# Entity data files
+
+Entity data files define the entity data to be indexed. Each row contains information for a single entity, defined in [JSON format](https://json.org/) and conforming to a specific [schema](reference-index-schema.md) with optional information about its relative probability among other entities.
+
+## Entity data JSON format
+
+Entity data is represented with JSON objects, with attributes values being represented by JSON name/value pairs conforming to the following rules:
+
+Attribute type | JSON value type | Example
+--- | --- | ---
+blob | string | `"Abstract": "It was the best of times, it was the worst of times ... It is a far, far better thing that I do, than I have ever done; it is a far, far better rest that I go to than I have ever known."`
+composite | object | `"F": { "FN": "machine learning", "FId": 119857082 }`
+date | string | `"D": "2015-05-18"`
+double | number | `"logprob": -2.71828182846`
+int32 | number | `"Y": 2015`
+int64 | number | `"Id": 8589934592`
+guid | string | `"Guid": "3d119f86-dcf0-4e35-9c01-b84832d851e9"`
+string | string | `"Ti": "an overview of microsoft academic service mas and applications"`
+
+### Multiple attribute values
+
+To define more than one value for an attribute you can either specify the attribute multiple times, e.g.:
+
+``` JSON
+{ "RId": 123, "RId": 456 }
+{ "F": { "FN": "machine learning", "FId": 119857082 }, "F": { "FN": "artificial intelligence", "FId": 154945302 } }
+```
+
+Or use JSON arrays, e.g.:
+
+``` JSON
+{ "RId": [ 123, 456 ] }
+{ "F": [ { "FN": "machine learning", "FId": 119857082 }, { "FN": "artificial intelligence", "FId": 154945302 } ] }
+```
+
+## Entity log probability
+
+An important component of the KES top-N retrieval algorithm is the concept of entity probability, which informs KES of each entities relative importance. Entity probability can be understood as the likelihood of it being named the most important in a survey conducted on all members its network/graph ([see "Saliency: An Eigencentrality Measure for Heterogeneous Dynamic Network" in "A Review of Microsoft Academic Services for Science of Science Studies"](https://www.frontiersin.org/articles/10.3389/fdata.2019.00045/pdf)).
+
+KES allows this to be defined for an entity as a relative log probability using the optional "logprob" attribute, which is an additional attribute that is always available and not defined in the [schema](reference-index-schema.md). Given a probability *p* between 0 and 1, the corresponding log probability can be computed as log(*p*), where log() is the natural log function. When no value is specified for logprob, the default value 0 is used.
```

Comment:
logprob -> weight

Reply:
KES allows you to define an entities "weight" using either a log probability score in the field "logprob" or a weight score derived from the log probability (-1000*logprob) in the field "weight"

We can also describe the "weight" field if you'd like, however I'd argue it's adding unneeded complexity

---
In reply to: [455289149](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/164#discussion_r455289149) [](ancestors = 455289149)

Reply:
We leveraged "Rank" provided by MAG as the "Weight" field in our subgraph sample script. We should be consistent with what we explain and what we use. Let's either modify the script and leverage "logprob" or illustrate "Weight" here.